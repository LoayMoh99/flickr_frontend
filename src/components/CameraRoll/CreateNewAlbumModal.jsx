import React,{useState , useEffect } from 'react';
import axios from "axios"
import './EditModal.css';
import './CreateNewAlbumModal.css';
const endpoint = 'http://localhost:3001/'

function CreateNewAlbumModal(props) {

  //new titles and description
  const [inputTitle, setInputTitle] = useState("");
  const [inputDescription , setInputDescription] = useState("");
  const [isNull,setIsNull]=useState(true);
  const { imgIdsCreateAlbum } = props; 
  const coverPhotoId = imgIdsCreateAlbum[0];
  console.log(props.imgIdsCreateAlbum);

  function handleTitleChange(event) {
    const newTitle = event.target.value;
    setInputTitle(newTitle);
    if(inputTitle==="")
    {
        setIsNull(true);
    }
    else{
        setIsNull(false);
    }
    console.log(inputTitle);
  }

  function handleDescriptionChange(event) {
    const newDescription  = event.target.value;
    setInputDescription(newDescription);
    console.log(inputDescription);
  }



//     const album = {
//       "title": inputTitle,
//       "description":inputDescription,
//       "photos":imgIdsCreateAlbum,
//       "coverPhoto":coverPhotoId
//        }


    const album = { 
        "id" : 29,
        "title": inputTitle,
        "description":inputDescription,
        "createdAt": "2021-03-02",
        "updatedAt": "2021-05-04",
        // "photos" : [
        //   {
        //       "photo_id" : 0,
        //       "photo_url" : "https://picsum.photos/300/200?random=1",
        //       "photo_owner_id": 0,
        //       "num_favs": 30,
        //       "num_views": 60,
        //       "num_comments":10,
        //       "photo_owner_name": "Khadija Khaled",
        //       "title": "First Photo Title",
        //       "privacy": true,
        //       "description": "First Photo description",
        //       "createdAt" : "2021-05-04"
        //   }
        // ],
        "photos":imgIdsCreateAlbum,
        //coverPhoto:"https://picsum.photos/300/200?random=1"
        coverPhoto:coverPhotoId
          
    };
    async function createAlbum (album){
    console.log(album);
    try{

        const response = await axios.post(endpoint + 'album', album);
        console.log("status");
        console.log(response.status);
    }catch (error){
        if (error.response){
          console.log(error.response.data);
          console.log(error.response.status);
          console.log(error.response.headers);
        } else if (error.request){
          console.log(error.request);
        } else {
          console.log('Error', error.message);
        }
        console.log(error);
    }
    }

    function create (){
        if(!isNull) //title is a required input can not be null
        {
            createAlbum(album);
        }
        props.onRequestCreateClose();
    }

  return (
    <>

      <div className="modal__backdrop_edit">
        <div className="modal__container_create">
          <span className="close_edit" onClick={props.onRequestCreateClose}>&times;</span>
          <h3 className="modal__title_edit">
            Create a new album
          </h3>
          {/* <div className="title-desc-container"> */}
         
            <input className="createTitle" type="text" placeholder="Album name" onChange={handleTitleChange} required/>
            {/* <hr /> */}
            <textarea className="createDescription" placeholder="Description (optional)" onChange={handleDescriptionChange} />
           
         
          
          <div className="createFooter">
            <button
              id="save_edit"
              type="button"
              onClick={create} 
            >
              Create
            </button>
            <button id="cancel_edit" type="button" onClick={props.onRequestCreateClose}>
              Cancel
            </button>
          </div>
         
        </div>
      </div>

    </>
  );
}

export default CreateNewAlbumModal;
