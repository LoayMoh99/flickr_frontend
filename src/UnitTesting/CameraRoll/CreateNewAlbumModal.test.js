import React from 'react';
import ReactDOM from 'react-dom';
import CreateNewAlbumModal from '../../components/CameraRoll/CreateNewAlbumModal';
import Enzyme, { shallow, mount } from 'enzyme';
import {render, fireEvent, cleanup} from '@testing-library/react';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() })

afterEach(cleanup)

it('CreateNewAlbumModal renders without crashing', ()=>{
    const dummy= ()=>{ return 1;}
    const div = document.createElement('div');
    ReactDOM.render(<CreateNewAlbumModal 
        onRequestCreateClose={dummy}
        imgIdsCreateAlbum= "[1,2]" 
       />, div);

});

it('CreateNewAlbumModal renders correctly enzyme', () => {
    const dummy= ()=>{ return 1;}
    const wrapper = shallow(<CreateNewAlbumModal 
        onRequestCreateClose={dummy}
        imgIdsCreateAlbum= "[1,2]"
       />)
       expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('Inputing Album name the state', () => {
        const dummy =()=>{return 1;};
        const { getByPlaceholderText} = render(<CreateNewAlbumModal 
            onRequestCreateClose={dummy}
            imgIdsCreateAlbum= "[1,2]"
           />);
    
        expect(getByPlaceholderText("Album name").value).toBe("")
    
        fireEvent.change(getByPlaceholderText("Album name"), {target: {value: 'Text' } } )
    
        expect(getByPlaceholderText("Album name").value).not.toBe("")
     })

     it('Inputing Album description the state', () => {
        const dummy =()=>{return 1;};
        const { getByPlaceholderText} = render(<CreateNewAlbumModal 
            onRequestCreateClose={dummy}
            imgIdsCreateAlbum= "[1,2]"
           />);
    
        expect(getByPlaceholderText("Description (optional)").value).toBe("")
    
        fireEvent.change(getByPlaceholderText("Description (optional)"), {target: {value: 'Text' } } )
    
        expect(getByPlaceholderText("Description (optional)").value).not.toBe("")
     })
    