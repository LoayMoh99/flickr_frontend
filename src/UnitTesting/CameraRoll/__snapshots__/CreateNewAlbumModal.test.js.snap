// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateNewAlbumModal renders correctly enzyme 1`] = `
<Fragment>
  <div
    className="modal__backdrop_edit"
  >
    <div
      className="modal__container_create"
    >
      <span
        className="close_edit"
        onClick={[Function]}
      >
        Ã—
      </span>
      <h3
        className="modal__title_edit"
      >
        Create a new album
      </h3>
      <input
        className="createTitle"
        onChange={[Function]}
        placeholder="Album name"
        required={true}
        type="text"
      />
      <textarea
        className="createDescription"
        onChange={[Function]}
        placeholder="Description (optional)"
      />
      <div
        className="createFooter"
      >
        <button
          id="save_edit"
          onClick={[Function]}
          type="button"
        >
          Create
        </button>
        <button
          id="cancel_edit"
          onClick={[Function]}
          type="button"
        >
          Cancel
        </button>
      </div>
    </div>
  </div>
</Fragment>
`;
